Firestore/Source/API/FIRAggregateQuery.mm

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FIRAggregateQuery+Internal.h"
+
+#import "FIRAggregateQuerySnapshot+Internal.h"
+#import "FIRQuery+Internal.h"
+
+#include "Firestore/core/src/api/aggregate_query.h"
+#include "Firestore/core/src/api/query_core.h"
+#include "Firestore/core/src/util/error_apple.h"
+#include "Firestore/core/src/util/statusor.h"
+
+@implementation FIRAggregateQuery {
+  FIRQuery *_query;
+  std::unique_ptr<api::AggregateQuery> _aggregation;
+}
+
+- (instancetype _Nonnull)initWithQuery:(FIRQuery *)query {
```

Comment:
I think we cannot do that because initWithQuery is designated initializer, see:

https://stackoverflow.com/questions/26185239/ios-designated-initializers-using-ns-designated-initializer

Reply:
I'm still a bit fuzzy on this (sorry, obj-c noob here). But the [explanation here](https://useyourloaf.com/blog/xcode-6-objective-c-modernization/#:~:text=A%20designated%20initializer%20must%20call%20(via%20super)%20a%20designated%20initializer%20of%20the%20superclass.%20Where%20NSObject%20is%20the%20superclass%20this%20is%20just%20%5Bsuper%20init%5D.) suggests that "A designated initializer must call (via super) a designated initializer of the superclass". ([example](https://stackoverflow.com/a/42064441))

Reply:
Yeah, you are right. Thanks!